medicines/api/src/document.rs

```diff
@@ -55,31 +39,229 @@ pagination! {Documents, DocumentEdge, Document}
 
 pub async fn get_documents(
     client: &impl Search,
-    search: Option<String>,
+    search: String,
     first: Option<i32>,
     last: Option<i32>,
     before: Option<String>,
     after: Option<String>,
-) -> Documents {
-    let placeholder_names: [&str; 1000] = ["A cool document"; 1000];
-    let edges = placeholder_names
+) -> Result<Documents, anyhow::Error> {
+    // TODO: Do something with last and after.
+
+    let offset = after.unwrap_or("-1".to_string()).parse::<i32>().unwrap() + 1;
+    let result_count = first.unwrap_or(10);
+
+    let azure_result = client
+        .search_with_pagination(
+            &search,
+            search_client::AzurePagination {
+                result_count: result_count,
+                offset: offset,
+            },
+            true,
+        )
+        .await?;
+
+    let mut cursor = offset.clone();
+    let edges = azure_result
+        .search_results
         .iter()
-        // .take(first as usize)
-        .map(|&name| Document::name_only(name.to_string()))
-        .map(|document| DocumentEdge {
-            node: document,
-            cursor: "cursor".to_owned(),
+        .map(|search_result| Document::from(search_result.clone()))
+        .map(|document| {
+            let edge = DocumentEdge {
+                node: document,
+                cursor: cursor.to_string(),
```

Comment:
As discussed with @sgrowe, these should be opaque. For now, we'll base 64 encode the page numbers so that our client can still jump to any page.